src/sentry/apidocs/extensions.py

```diff
@@ -16,3 +22,44 @@ def get_security_requirement(self, auto_schema):
                 permissions.add(p)
 
         return {self.name: list(permissions)}
+
+
+# TODO: map things in registry?
+# add docstring as description
+# excluded attributes?
+# optional/required in responses
+class SentryResponseSerializerExtension(OpenApiSerializerExtension):  # type: ignore
+    priority = 0
+    target_class = "sentry.api.serializers.base.Serializer"
+    match_subclasses = True
+
+    def get_name(self) -> Optional[str]:
+        name: str = self.target.__name__
+        return name
+
+    def map_serializer(self, auto_schema: AutoSchema, direction: Direction) -> Any:
+        serializer_signature = inspect.signature(self.target.serialize)
+        return resolve_type_hint(serializer_signature.return_annotation)
+
+
+# class SentryInlineResponseSerializerExtension(OpenApiSerializerExtension):  # type: ignore
+#     priority = 0
+#     target_class = "sentry.apidocs.schemaserializer.RawSchema"
+#     match_subclasses = True
+
+#     def get_name(self) -> Optional[str]:
+#         name: str = self.target.__name__
+#         return name
+
+#     def map_serializer(self, auto_schema: AutoSchema, direction: Direction) -> Any:
+#         return resolve_type_hint(self.target.typeSchema)
+
+
+class RawSchema:
+    def __init__(self, t: type) -> None:
+        self.typeSchema = t
+
+
+def inline_sentry_response_serializer(name: str, t: type) -> type:
```

Comment:
this function would be used where e.g. we are returning a list of items instead of a singular item. for the index GET corresponding to the SCIM details, we'd have `
```
inline_sentry_response_serializer(
    'SCIMMemberIndexResponse',
    List[OrganizationMemberSCIMSerializerResponse]
)
```
as the response schema.